biz.aQute.tester.junit-platform/src/aQute/tester/bundle/engine/discovery/BundleSelectorResolver.java

```diff
@@ -389,7 +395,8 @@ private void markClassesResolved(Bundle bundle) {
 						}
 						return selectMethod(testClass, DiscoverySelectors.selectMethod(testcase));
 					}
-				} catch (ClassNotFoundException cnfe) {
+				} catch (ClassNotFoundException | NoClassDefFoundError cnfe) {
```

Comment:
Putting `NoClassDefFoundError` here can hide our own stupidity in other parts of this code :-( Perhaps we should narrow the window for catching `NoClassDefFoundError` to only the calls to testClass.getDeclaredMethods() and	testClass.getDeclaredFields() since I think that is the only source for which we need to worry about it?

Reply:
Fair point. Note that `loadClass()` can throw `NoClassDefFoundError` too if the superclass doesn't resolve; see here: 
https://github.com/bndtools/bnd/blob/5693958978d52d83cf847a649af841cede89cc00/biz.aQute.tester.test/test/aQute/tester/bundle/engine/test/BundleEngineTest.java#L1227-L1244

So I have included the `loadClass()` in the scope of the catch for `NoClassDefFoundError` too.